src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -33,6 +34,7 @@ final class NodeFactory
         'DatePublished' => SortClause\DatePublished::class,
         'ContentName' => SortClause\ContentName::class,
     ];
+    private const MAX_AGGREGATED_LOCATION_IDS = 100;
```

Comment:
I think this should be configurable if there is possibility to fine tune performance on different setups.